bundles/GraphicServiceOrder/Service/OrderService.php

```diff
@@ -47,16 +50,76 @@ public function __construct(
         GsOrderRepository $gsOrderRepository,
         KernelInterface $appKernel,
         MessageBusInterface $messageBus,
-        string $ownCloudFilesFolder
+        string $ownCloudFilesFolder,
+        TokenStorageInterface $tokenStorage
     ) {
         $this->entityManager = $entityManager;
-        $this->gsOrderRepository = $gsOrderRepository;
         $this->hammerService = $hammerService;
         $this->ownCloudService = $ownCloudService;
         $this->gsOrderRepository = $gsOrderRepository;
         $this->appKernel = $appKernel;
         $this->messageBus = $messageBus;
         $this->ownCloudFilesFolder = $ownCloudFilesFolder;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * Preset some values from user entity.
+     *
+     * @return \GraphicServiceOrder\Entity\GsOrder
+     */
+    public function prepareOrder()
+    {
+        $gsOrder = new GsOrder();
+        $token = $this->tokenStorage->getToken();
+        if (null !== $token) {
+            $user = $token->getUser();
+            $gsOrder->setFullName($user->getFullName());
+            $gsOrder->setAddress($user->getAddress());
+            $gsOrder->setDepartment($user->getDepartment());
+            $gsOrder->setPostalcode($user->getPostalCode());
+            $gsOrder->setCity($user->getCity());
+        }
+
+        return $gsOrder;
+    }
+
+    /**
+     * Fetch user mail from active user.
+     *
+     * @return string
+     */
+    public function getUserEmail()
+    {
+        $token = $this->tokenStorage->getToken();
+        if (null !== $token) {
+            $user = $token->getUser();
+
+            return $user->getEmail();
+        }
+
+        return '';
+    }
+
+    /**
+     * Update active user with submitted values.
+     *
+     * @param $gsOrder
+     */
+    private function updateUser($gsOrder)
+    {
+        $token = $this->tokenStorage->getToken();
+        if (null !== $token) {
+            $user = $token->getUser();
+            $user->setFullName($gsOrder->getFullName());
+            $user->setDepartment($gsOrder->getDepartment());
+            $user->setAddress($gsOrder->getAddress());
+            $user->setPostalCode($gsOrder->getPostalcode());
+            $user->setCity($gsOrder->getCity());
+
+            $this->entityManager->persist($user);
```

Comment:
All updates to `User` should be handled by the `updateUser` function in the `UserManager` service.

Reply:
Changed